classes/local/broker.php

```diff
@@ -45,243 +45,6 @@
-        if ($recordings[$params['id']]->imported) {
-            /* Since the recording link is the one fetched from the BBB server, imported recordings can not be
-             * unprotected. There is no need to do anything else.
-             */
-            return array('status' => 'This action can not be performed on imported links.');
-        }
-        // As the recordingid was not identified as imported recording link, execute unprotect on a real recording.
-        return array(
-            'status' => recording_proxy::bigbluebutton_update_recordings(
-                $params['id'],
-                array('protect' => 'false')
-            )
-        );
-    }
-
-    /**
-     * Helper for performing delete on recordings.
-     *
-     * @param array $params
-     * @param array $recordings
-     *
-     * @return array
-     */
-    public static function recording_action_delete($params, $recordings) {
-        if ($recordings[$params['id']]->imported) {
-            // Execute delete on imported recording link.
-            return array(
-                'status' => recording::delete(
-                    $recordings[$params['id']]->id
-                )
-            );
-        }
-        // As the recordingid was not identified as imported recording link, execute delete on a real recording.
-        // Step 1, delete imported links associated to the recording.
-        recording::delete_by(
-            ['recordingid' => $recordings[$params['id']]->recordingid,
-            'imported' => true]
-        );
-        // Step 2, perform the actual delete by sending the corresponding request to BBB.
-        return array(
-            'status' => recording_proxy::bigbluebutton_delete_recordings($params['id'])
-        );
-    }
-
-    /**
-     * Helper for performing edit on recordings.
-     *
-     * @param array $params
-     * @param array $recordings
-     *
-     * @return array
-     */
-    public static function recording_action_edit($params, $recordings) {
-        if ($recordings[$params['id']]->imported) {
-            // Execute update on imported recording link.
-            return array(
-                'status' => recording_proxy::bigbluebutton_update_recording_imported(
-                    $recordings[$params['id']]['imported'],
-                    json_decode($params['meta'], true)
-                )
-            );
-        }
-
-        // As the recordingid was not identified as imported recording link, execute update on a real recording.
-        return array(
-            'status' => recording_proxy::bigbluebutton_update_recordings(
-                $params['id'],
-                json_decode($params['meta'])
-            )
-        );
-    }
-
-    /**
-     * Helper for responding when recording ready is performed.
-     *
-     * @param array $params
-     * @param object $bigbluebuttonbn
-     *
-     * @return void
-     */
-    public static function recording_ready($params, $bigbluebuttonbn) {
```

Comment:
The response can not just disappear. We need to have a way to notify BBB what happened with the callback.